glean-core/android/src/main/java/mozilla/telemetry/glean/Glean.kt

```diff
@@ -102,15 +102,19 @@ open class GleanInternalAPI internal constructor () {
      * @param applicationContext [Context] to access application features, such
      * as shared preferences
      * @param configuration A Glean [Configuration] object with global settings.
+     * @param uploadEnabled A [Boolean] that determines the initial state of the uploader
      */
     @Suppress("ReturnCount", "LongMethod", "ComplexMethod")
     @JvmOverloads
     @Synchronized
     @MainThread
     fun initialize(
         applicationContext: Context,
-        configuration: Configuration = Configuration()
+        configuration: Configuration = Configuration(),
+        uploadEnabled: Boolean
```

Comment:
Since we're not making `uploadEnabled` optional, I think we need to move it before `configuration`: otherwise we make it mandatory to provide `Configuration()` even when not required by the embedder.

Reply:
It's not mandatory, you just have to use the label.  I don't have any preference to the order though and you are right, it's usually better to put parameters without defaults before those that have them (thinking of languages where that's a hard rule)